### PR TITLE
fix(Core): macOS build

### PR DESCRIPTION
**Pull request by FrancescoBorzi**

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -4,11 +4,11 @@
 * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
 */
 
+#include <inttypes.h>
```
Comment (Kitzunu) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:7]: shouldnt < > includes be at the bottom?

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 2]
```diff
@@ -1206,7 +1206,7 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                     break;
 
                 instance->SetGuidData(e.action.setInstanceData64.field, targets->front()->GetGUID());
-                LOG_DEBUG("sql.sql", "SmartScript::ProcessAction: SMART_ACTION_SET_INST_DATA64: Field: %u, data: %lu",
+                LOG_DEBUG("sql.sql", "SmartScript::ProcessAction: SMART_ACTION_SET_INST_DATA64: Field: %u, data: %" PRIu64,
```
Comment (Winfidonarleyan) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:1209]: https://github.com/azerothcore/azerothcore-wotlk/blob/5787d00d545aaf4139f84f3e207c3b69b5b4231f/src/common/Define.h#L105

File: src/server/game/AI/SmartScripts/SmartScript.cpp (hunk 1 above)
Comment (Winfidonarleyan) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:7]: No need this

**Review by Kitzunu — APPROVED** [review]

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 3]
```diff
@@ -1206,7 +1205,7 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                     break;
 
                 instance->SetGuidData(e.action.setInstanceData64.field, targets->front()->GetGUID());
-                LOG_DEBUG("sql.sql", "SmartScript::ProcessAction: SMART_ACTION_SET_INST_DATA64: Field: %u, data: %lu",
+                LOG_DEBUG("sql.sql", "SmartScript::ProcessAction: SMART_ACTION_SET_INST_DATA64: Field: %u, data: " SI64FMTD,
```
Comment (Winfidonarleyan) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:1208]:
```suggestion
                LOG_DEBUG("sql.sql", "SmartScript::ProcessAction: SMART_ACTION_SET_INST_DATA64: Field: %u, data: " SZFMTD,
```

Example 
```cpp
LOG_INFO("server.loading", ">> Loaded " SZFMTD " broadcast texts in %u ms", _broadcastTextStore.size(), GetMSTimeDiffToNow(oldMSTime));
```

Comment (FrancescoBorzi) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:1208]: @Winfidonarleyan why? that would not work:

image

Comment (Winfidonarleyan) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:1208]: Then how does this work
```cpp
LOG_INFO("server.loading", ">> Loaded " SZFMTD " broadcast texts in %u ms", _broadcastTextStore.size(), GetMSTimeDiffToNow(oldMSTime));
```

Comment (Winfidonarleyan) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:1208]: Yes, okay, my bad

**Pull request merged**